packages/server-side-graphql-client/lib/server-side-graphql-client.js

```diff
@@ -84,31 +84,39 @@ const getItem = async ({ keystone, listKey, itemId, returnFields = `id`, context
 const getItems = async ({
   keystone,
   listKey,
+  first,
+  skip,
   where = {},
   pageSize = 500,
   returnFields = `id`,
+  sortBy = 'id_ASC',
   context,
 }) => {
-  const { listQueryName, whereInputName } = keystone.lists[listKey].gqlNames;
-  const query = `query ($first: Int!, $skip: Int!, $where: ${whereInputName}) { ${listQueryName}(first: $first, skip: $skip, where: $where) { ${returnFields} }  }`;
+  const { listQueryName, whereInputName, listSortName } = keystone.lists[listKey].gqlNames;
+  const query = `query ($first: Int!, $skip: Int!, $sortBy: [${listSortName}!], $where: ${whereInputName}) { ${listQueryName}(first: $first, skip: $skip, sortBy: $sortBy, where: $where) { ${returnFields} }  }`;
 
-  let skip = 0;
   let latestResult;
+  let _skip = skip || 0;
+  let _first = first || pageSize;
   const allItems = [];
 
   do {
     const response = await runQuery({
       keystone,
       query,
       context,
-      variables: { first: pageSize, skip, where },
+      variables: { first: _first, skip: _skip, sortBy, where },
```

Comment:
I don't think this works as expected.

If I ask for `first: 100` with `pageSize: 10` then each query should be pulling `10` items at a time until we have collected a total of 100 items.

The current implementation will completely override all pagination if `first` is supplied.

Reply:
We also need to be careful not to overfetch. If I have `first: 101, pageSize: 10` we need to be careful not to pull down `110` items.

Reply:
Oh, nice catch. Thank you.